File: src/main/java/io/github/kingg22/api/vacunas/panama/service/IPersonaService.java
package io.github.kingg22.api.vacunas.panama.service;

import io.github.kingg22.api.vacunas.panama.persistence.entity.Persona;
import jakarta.validation.constraints.NotNull;
import java.util.Optional;
import java.util.UUID;

public interface IPersonaService {
    Optional<Persona> getPersona(@org.jetbrains.annotations.NotNull @NotNull String identifier);

    Optional<Persona> getPersonaByUserID(@org.jetbrains.annotations.NotNull @NotNull UUID idUser);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/service/IPersonaService.kt
package io.github.kingg22.api.vacunas.panama.service

import io.github.kingg22.api.vacunas.panama.persistence.entity.Persona
import jakarta.validation.constraints.NotNull
import java.util.Optional
import java.util.UUID

interface IPersonaService {
    fun getPersona(identifier: @NotNull String): Optional<Persona>

    fun getPersonaByUserID(idUser: @NotNull UUID): Optional<Persona>
}
